frontend/catalyst/compiler.py

```diff
@@ -268,6 +270,60 @@ def get_output_filename(infile):
         return infile.replace(".ll", ".o")
 
 
+class WrapperToCatchExceptions(PassPipeline):
```

Comment:
Btw would it make sense to move this class to the `catalyst.utils` module where our other wrappers already live, like the `init` and `finialize` functions?

Reply:
Yes, I am still working on how I would rather fit this in. Hence the in-progress pull request.

Reply:
The main reason why I see `WrapperToCatchExceptions` placed here as opposed to `catalyst.utils` is that it includes more logic than the other templates. And it is a well defined compiled object. As in, the others are part of some module, but this one is its own "file" that will be compiled.

Reply:
> As in, the others are part of some module, but this one is its own "file" that will be compiled.

This to me would indicate that it would do well in a separate module to not detract from the main purpose of this module, which is to provide the infrastructure for and run the compilation stack, rather than _defining_ a standalone program that is separately compiled. But it's not a necessity.

What about my comment on the inheritance on `PassPipeline`? Your last comment seems to be in agreement that `WrapperToCatchException` is definitely not a compilation pass applied to the user program, but rather another program linked into it.

Reply:
Is linking not part of the compilation pipeline? :sweat_smile: I think it is splitting hairs a bit. I think just looking at it locally, it makes sense to place it here as it is dependent on the number of inputs and on the name of the function and it will be used as an object to be linked against. Moving it away, hides it.

Reply:
Yup I'm fine leaving it here 👍 

But `WrapperToCatchException` is not a `PassPipeline`, from an OO point of view that makes no sense.

Reply:
Ok, will change!